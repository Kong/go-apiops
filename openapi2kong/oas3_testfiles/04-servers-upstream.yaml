# if the servers block has more than 1 url, then an upstream
# is generated with a target for each sesrver

openapi: '3.0.0'
info:
  title: Simple API overview
  version: v2
servers:
  - url: https://server1.com/
  - url: https://server2.com/
paths:
  /:
    get:
      operationId: OpsId
      summary: List API versions
      responses:
        '200':
          description: |-
            200 response
