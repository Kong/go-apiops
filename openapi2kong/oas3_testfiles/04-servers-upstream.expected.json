{
  "_format_version": "3.0",
  "services": [
    {
      "host": "simple-api-overview.upstream",
      "id": "0907c4ab-d9e4-5d21-813b-c57a97eeaad9",
      "name": "simple-api-overview",
      "path": "/",
      "plugins": [],
      "port": 443,
      "protocol": "https",
      "routes": [
        {
          "id": "eee036de-517e-59cf-a2e0-17b3adfa31b5",
          "methods": [
            "GET"
          ],
          "name": "simple-api-overview_opsid",
          "paths": [
            "~/$"
          ],
          "plugins": [],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_04-servers-upstream.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_04-servers-upstream.yaml"
      ]
    }
  ],
  "upstreams": [
    {
      "id": "811c42d6-ef18-5296-a550-7dca2262b4d8",
      "name": "simple-api-overview.upstream",
      "tags": [
        "OAS3_import",
        "OAS3file_04-servers-upstream.yaml"
      ],
      "targets": [
        {
          "tags": [
            "OAS3_import",
            "OAS3file_04-servers-upstream.yaml"
          ],
          "target": "server1.com:443"
        },
        {
          "tags": [
            "OAS3_import",
            "OAS3file_04-servers-upstream.yaml"
          ],
          "target": "server2.com:443"
        }
      ]
    }
  ]
}